catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java

```diff
@@ -76,42 +79,179 @@ public UtmUpsCoordinateProcessor(CoordinateSystemTranslator translator) {
    */
   public void enhanceResults(final List<Suggestion> results, final String query) {
     LOGGER.trace("(UTM/UPS) Adding result for query [{}]", query);
-    final LiteralSuggestion literal = getUtmUpsSuggestion(query);
-    if (literal != null && literal.hasGeo()) {
-      LOGGER.trace("Adding the UTM/UPS suggestion to results [{}]", literal);
-      results.add(0, literal);
-    }
+    final List<LiteralSuggestion> literals = getUtmUpsSuggestions(query);
+    literals
+        .stream()
+        .filter(Objects::nonNull)
+        .filter(LiteralSuggestion::hasGeo)
+        .forEach(
+            literal -> {
+              LOGGER.trace("Adding the UTM/UPS suggestion to results [{}]", literal);
+              results.add(0, literal);
+            });
     LOGGER.trace("(UTM/UPS) Done");
   }
 
   @Nullable
```

Comment:
Should we keep this annotation on the method?

Reply:
nice catch, I removed it